File: pawtient-care-be/src/main/java/com/aksoyakin/pawtientcarebe/service/pet/PetService.java
package com.aksoyakin.pawtientcarebe.service.pet;

import com.aksoyakin.pawtientcarebe.model.Pet;

import java.util.List;

public interface PetService {

    List<Pet> savePetForAppointment(List<Pet> pets);

    Pet updatePet(Pet pet, Long petId);

    void deletePet(Long petId);

    Pet getPetById(Long petId);

    List<String> getPetTypes();

    List<String> getPetColors();

    List<String> getPetBreeds(String petType);
}

File: pawtient-care-fe/src/components/pet/PetService.js
import {api} from "../utils/api.js";

export async function getPetTypes() {
    try {
        const result = await api.get("/pets/get-types")
        return result.data
    } catch (error) {
        throw error;
    }
}

export async function getPetColors() {
    try {
        const result = await api.get("/pets/get-pet-colors")
        return result.data
    } catch (error) {
        throw error;
    }
}

export async function getPetBreeds(petType) {
    try {
        const result = await api.get(`/pets/get-pet-breeds?petType=${petType}`);
        return result.data
    } catch (error) {
        throw error;
    }
}